modules/data.remote/R/call_remote_process.R

```diff
@@ -0,0 +1,400 @@
+##' call the Python - remote_process from PEcAn and store the output in BETY
+##' 
+##' @name call_remote_process
+##' @title call_remote_process
+##' @export
+##' @param settings PEcAn settings list containing remotedata tags: siteid, sitename, raw_mimetype, raw_formatname, geofile, outdir, start, end, source, collection, scale, projection, qc, algorithm, credfile, pro_mimetype, pro_formatname, out_get_data, out_process_data
+##' @examples
+##' \dontrun{
+##' call_remote_process(settings)
+##' }
+##' @author Ayush Prasad
+##'
+
+call_remote_process <- function(settings){
+  
+  # information about the date variables used in call_remote_process -
+  # req_start, req_end : start, end dates requested by the user, the user does not be aware about the status of the requested file in the DB  
+  # start, end : effective start, end dates created after checking the DB status. These dates are sent to remote_process for downloading and processing data
+  # write_raw_start, write_raw_end : start, end dates which are used while inserting and updating the DB
+  # the "pro" version of these variables have the same meaning and are used to refer to the processed file
+
+  reticulate::import_from_path("remote_process", file.path("..", "inst"))
+  reticulate::source_python(file.path("..", "inst", "remote_process.py"))
+  
+  input_file <- NULL
+  stage_get_data <- NULL
+  stage_process_data <- NULL
+  raw_merge <- NULL
+  pro_merge <- NULL
+  existing_raw_file_path <- NULL
+  existing_pro_file_path <- NULL
+  
+  # extract the variables from the settings list
+  siteid <- settings$remotedata$siteid
+  sitename <- settings$remotedata$sitename
+  raw_mimetype <- settings$remotedata$raw_mimetype
+  raw_formatname <- settings$remotedata$raw_formatname
```

Comment:
Things like mimetype and formatname are going to be fixed for a specific data source, and thus shouldn't need to be provided by the user (this would just be a potential source of error and makes it harder for users as they have to look up info that only has one 'right' answer). In met we solve this problem by putting simple "registry" metadata files for each data type/source in the package itself.

Reply:
Also, if you're adding a bunch of stuff to the Settings, you definitely need to update the PEcAn documentation to document any new settings options.

Reply:
Yes, I agree with Mike that we won't be needing mimetypes and formats specified in the settings tags, I'll give Ayush's code a try and see if we can have registry files or some lookup table sort of thing for this